server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -3237,13 +3245,29 @@ protected void runInContext() {
         }
     }
 
+    private void cleanupDownloadUrls(VMInstanceVO systemVm) {
```

Comment:
@ravening DatacenterId should be ok here, to cleanup urls.

Reply:
Yes but I didnt pass it purposefully because in case in future we need to do further processing with the systemvm, we will have entire data. If I just pass data center id, we dont have reference to systemvm

Reply:
@ravening in case, any data is required for further processing in future in this method, the method definition/implementation can be updated based on the requirement in future. As the datacenter is only being used to cleanup the urls, so better use the datacenter id for the cleanup purpose here, and update the method name accordingly (suggestion below).

```suggestion
    private void cleanupDownloadUrlsInZone(long zoneId) {
```

Reply:
ping @ravening. Can you answer @sureshanaparti here?

Reply:
ping @ravening , please update on this. thanks.

Reply:
> ping @ravening , please update on this. thanks.

@sureshanaparti i would rather prefer to keep the method definition as is rather than changing it again future which I double work

Reply:
@ravening the download urls are related to zone, so I recommend to go with zone id, which is relavent. Passing _systemVm_ (VMInstanceVO of SSVM) is irrelevant here. Not sure what will be the future double work you are pointing to.